internal/pkg/vm/actor/builtin/power/power_test.go

```diff
@@ -0,0 +1,190 @@
+
+		// set power again
+		report2 := types.NewPowerReport(77, 990)
+		pdata2 := actor.MustConvertParams(report2, minerAddr)
+		msg2 := types.NewUnsignedMessage(address.TestAddress, address.PowerAddress, 0, hundredAttoFIL, power.ProcessPowerReport, pdata2)
+		result2, err := th.ApplyTestMessage(st, vms, msg2, types.NewBlockHeight(0))
+		require.NoError(t, err)
+		require.Nil(t, result2.ExecutionError)
+
+	})
+
+	t.Run("process nonexistent account", func(t *testing.T) {
+		st, vms := th.RequireCreateStorages(ctx, t)
+		report := types.NewPowerReport(1, 2)
+		pdata := actor.MustConvertParams(report, address.TestAddress)
+		msg := types.NewUnsignedMessage(address.TestAddress, address.PowerAddress, 0, hundredAttoFIL, power.ProcessPowerReport, pdata)
+		result, err := th.ApplyTestMessage(st, vms, msg, types.NewBlockHeight(0))
+		assert.NoError(t, err)
+		assert.Equal(t, power.Errors[power.ErrUnknownEntry], result.ExecutionError)
+	})
+}
+
+func TestGetTotalPower(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	hundredAttoFIL := types.NewAttoFILFromFIL(100)
+
+	st, vms := th.RequireCreateStorages(ctx, t)
+	pid := th.RequireRandomPeerID(t)
+	minerAddr := requireCreateMiner(hundredAttoFIL, t, st, vms, address.TestAddress, pid, 0)
+
+	// set power
+	report1 := types.NewPowerReport(600, 400)
+	pdata1 := actor.MustConvertParams(report1, minerAddr)
+	msg1 := types.NewUnsignedMessage(address.TestAddress, address.PowerAddress, 0, hundredAttoFIL, power.ProcessPowerReport, pdata1)
+	result1, err := th.ApplyTestMessage(st, vms, msg1, types.NewBlockHeight(0))
+	require.NoError(t, err)
+	require.Nil(t, result1.ExecutionError)
+
+	// get power
+	msg2 := types.NewUnsignedMessage(address.TestAddress, address.PowerAddress, 0, hundredAttoFIL, power.GetTotalPower, nil)
+	result2, err := th.ApplyTestMessage(st, vms, msg2, types.NewBlockHeight(1))
+	require.NoError(t, err)
+	require.Equal(t, 1, len(result2.Receipt.Return))
+	totalPowerIface, err := abi.Deserialize(result2.Receipt.Return[0], abi.BytesAmount)
+	require.NoError(t, err)
+	totalPower, ok := totalPowerIface.Val.(*types.BytesAmount)
+	require.True(t, ok)
+	assert.Equal(t, types.NewBytesAmount(1000), totalPower)
+}
+
+func TestRemoveStorageMiner(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	hundredAttoFIL := types.NewAttoFILFromFIL(100)
+	t.Run("happy path", func(t *testing.T) {
+		st, vms := th.RequireCreateStorages(ctx, t)
+		pid := th.RequireRandomPeerID(t)
+		minerAddr := requireCreateMiner(hundredAttoFIL, t, st, vms, address.TestAddress, pid, 0)
+
+		pdata := actor.MustConvertParams(minerAddr)
+		msg := types.NewUnsignedMessage(address.TestAddress, address.PowerAddress, 0, hundredAttoFIL, power.RemoveStorageMiner, pdata)
+		result, err := th.ApplyTestMessage(st, vms, msg, types.NewBlockHeight(0))
+		assert.NoError(t, err)
+		assert.Nil(t, result.ExecutionError)
```

Comment:
Please confirm that the power table entry has been removed. There are several ways this could be done: checking total power, checking for errors when updating or removing an entry, direct state inspection. I would consider fixing the glaring omission in this actor and add an un-speced method to retrieve a power table entry.

Reply:
Thanks, I'll see if FakeVMContext will help me here.

Reply:
yeah also we're going to need that for any power table integration so maybe now's the time...